spec/ics-027-direct-interchain-exchange/ReadMe.md

```diff
@@ -0,0 +1,74 @@
+# Cosmos Direct Interchain Exchange Protocol (DIEP)
+
+
+#### Core Idea:
+
+Asynchronous liquidity available with IBC that is resistant to theft by misbehaving validators. This eliminates the need to concentrate liquidity (limit orders) in a third party blockchain/venue. Price discovery and execution becomes native between Cosmos zones (e.g. N^2 IBC connections for N zones). There is a cost in terms of higher latency from this architecture.
+
+Exchange consists of three core components.
+
+1. Commitment of Funds for an order
+2. Order matching
+3. Execution and delivery
+
+A DEX chain like Binance combines these 3 functions into a single synchronous environment. This comes with risks including asset peg risks, and trusting the validator set of faithfully execute your order.	
+
+We propose an alternative model.  In this model, each of the 3 phases is asynchronously executed on independent blockchains.
+
+#### Cast of Characters
+
+Alice has Foo tokens on FooChain.
+
+Bob has Bar tokens on BarChain.
+
+FooChain and BarChain are connected via IBC.
+
+There exists a Hub chain.  FooChain and BarChain are also connected via IBC to the HubChain.
+
+Alice perceives the security of FooChain and Hub chain to be *strong*. 
+Alice perceive the security of BarChain to be /moderate/.
+Or, Alice perceives the security of both FooChain and BarChain to be *strong*.
+
+Bob perceives the security of Bar Chain and Hub chain to be *strong*.
+Bob perceive the security of Foo Chain to be /moderate/.
+Or, Bob perceives the security of both FooChain and BarChain to be *strong*.
+
+#### Protocol
+
+For simplicity, we will describe only one scenario, where Alice wishes to submit a limit order to purchase Bar tokens with her Foo tokens, where the desired price is lower than market (e.g. the limit order will not be matched immediately).
+
+Alice submits to FooChain a limit order transaction on FooChain: {type:"limit", pair:"bar/foo", sell:"20foo", price:"0.9"}
+
+FooChain sees that the price of "0.9" is lower than the current price of "1", so it locks Alice's 20 Foo tokens, and sends this order to BarChain via IBC.
```

Comment:
The price on BarChain? Where is this price stored, and what if FooChain's data is out of date?

Reply:
i agree that this analogy is inaccurate. there exists no unique "price" in multi-exchange environment. 

also, matching on blockchains are not continues but descrete process. therefore, whether the order is immediately matched or not cannot be decided by recent price.

so, IBC should always assume "no-immediate-matching" condition from order step. the protocol does not know whether it is immediately matched or not until actual matching happens.